contrib/auto-render/auto-render.js

```diff
@@ -55,7 +55,17 @@ const renderElem = function(elem, optionsCopy) {
         const childNode = elem.childNodes[i];
         if (childNode.nodeType === 3) {
             // Text node
-            const frag = renderMathInText(childNode.textContent, optionsCopy);
+            // Concatenate all sibling text nodes.
+            // Webkit browsers split very large text nodes into smaller ones,
+            // so the delimiters may be split across different nodes.
+            let textContentConcat = childNode.textContent;
+            let sibling = childNode.nextSibling;
+            while (sibling && (sibling.nodeType == 3)) {
+                textContentConcat += sibling.textContent;
+                sibling.remove();
```

Comment:
This looks like it will remove most text nodes that have no math in them; notice the `if (frag)` below.

Reply:
Thank you! You are right, if Chrome, etc., encountered a large text node, that would happen. I tried to address your comment.

As for efficiency, I'm wondering if testing if `childNode.textContent` contains any of the left delimiters first would improve the algorithm.

I attached an example that renders fine on Firefox but not on Chrome or Edge.

[katex_large_math_mwe.zip](https://github.com/KaTeX/KaTeX/files/7573075/katex_large_math_mwe.zip)

Reply:
> As for efficiency, I'm wondering if testing if childNode.textContent contains any of the left delimiters first would improve the algorithm.

Probably... but I like the simplicity of this strategy (for now), and it only impacts the performance when there are multiple sibling text nodes. If we wanted to be efficient, we'd only work with two siblings at once (or more if a math section actually spans more than two text nodes), but that's quite a bit more complicated and probably best left for a future PR if it ends up being an issue for someone.

> I attached an example that renders fine on Firefox but not on Chrome or Edge.

Thanks! I confirmed this.

I also confirmed that your changes fix the problem on Chrome (by running `yarn build` and using the resulting `auto-render.js` instead).